src/reasoning/PrologClient.h

```diff
@@ -18,18 +18,41 @@ class PrologClient {
     private:
         std::string _class;
         std::string _name;
+        //std::string _timeStamp;
 
     public:
         Instance(const std::string& classType, std::string name);
         const std::string& Get_class() const noexcept;
         uint Get_id() const noexcept;
     };
-
+    struct Property{
+        private:
+            std::string _name;
+            std::string _value;
+        public:
+        Property(std::string name, std::string  value);
+        const std::string& Get_name() const noexcept;
+    }
     PrologClient();
 
     uint CreateTypeInstance(const std::string &associatedClass);
 
     void test_prolog_query();
+    void create_instance (const Instance& instance);
+    void assert_property(const Instance& instance,const Property& property);
+    void assert_timeStamp_property(const Instance& instance,const Instance& timeStamp);
+    void assert_CardPosition_property(const Instance& instance,const Instance& position);
+    void delete_instance(const Instance& instance);
+    void save_action_TurnCard(const TurnCard& action,uint timeInstant);
```

Comment:
you can give every `save_action_...(...)` method the same name (`save_action(...)`

Reply:
all the actions are already labeled as save_action
The rest are complement functions